mailet/standard/src/main/java/org/apache/james/transport/mailets/AddFooter.java

```diff
@@ -94,11 +94,18 @@ private boolean attachFooter(MimePart part) throws MessagingException, IOExcepti
         if (part.isMimeType("multipart/mixed")
                 || part.isMimeType("multipart/related")) {
             MimeMultipart multipart = (MimeMultipart) part.getContent();
-            return attachFooterToFirstPart(multipart);
+            boolean added = attachFooterToFirstPart(multipart);
+            if (added) {
+            	part.setContent(multipart);
+            }
+            return added;
 
         } else if (part.isMimeType("multipart/alternative")) {
             MimeMultipart multipart = (MimeMultipart) part.getContent();
-            return attachFooterToAllSubparts(multipart);
+            boolean added = attachFooterToAllSubparts(multipart);
+            if (added) {
+            	part.setContent(multipart);
+            }
```

Comment:
I guess you forgot the `return added` stuff here...

That would cause `multipart/alternative` to not be well handled anymore...

By the way forgetting this return breaks tests.